piqueserver/player.py

```diff
@@ -88,8 +91,6 @@ def on_login(self, name: str) -> None:
         self.protocol.irc_say('* %s (IP %s, ID %s) entered the game!' %
                               (self.name, self.address[0], self.player_id))
         if self.user_types is None:
```

Comment:
Change to `if not self.user_types:` since it might be an empty set rather than None.

Reply:
Hm, you're right, we have a problem here. But it's a different one. `on_join` is called multiple times for the lifetime of a connection, once per match. We probably don't want to login the users as admin every time when a new match starts. But we need the user to be joined to be able to execute `on_user_login` as it might send chat messages.